middleware/session/session.go

```diff
@@ -171,6 +171,11 @@ func (s *Session) setCookie() {
 
 	s.ctx.Response().Header.SetCookie(fcookie)
 	fasthttp.ReleaseCookie(fcookie)
+
+	if s.config.Source == SourceHeader {
+		s.ctx.Request().Header.SetBytesV(s.config.CookieName, []byte(s.id))
+		s.ctx.Response().Header.SetBytesV(s.config.CookieName, []byte(s.id))
+	}
```

Comment:
The code above this if statement will still execute even if the source is a header. It should be in an if statement that checks the source.